grails-app/controllers/org/broadinstitute/orsp/SubmissionController.groovy

```diff
@@ -55,48 +48,61 @@ class SubmissionController extends AuthenticatedController {
-        render(view: '/submission/submission',
-                model: [issue:                      issue,
-                        submission:                 submission,
-                        docTypes:                   PROJECT_DOC_TYPES,
-                        submissionTypes:            submissionTypes,
-                        submissionNumberMaximums:   submissionNumberMaximums,
-                        defaultType:                defaultType
-                ])
+        render( [issue:                      issue,
+                 typeLabel:                  issue.typeLabel,
+                 submission:                 submission,
+                 documents:                  submission?.documents,
+                 docTypes:                   PROJECT_DOC_TYPES,
+                 submissionTypes:            submissionTypes,
+                 submissionNumberMaximums:   submissionNumberMaximums,
+                 defaultType:                defaultType
+        ] as JSON)
     }
 
     def save() {
-        Issue issue = queryService.findByKey(params.projectKey)
-        Submission submission
-        if (params?.submissionId) {
-            submission = Submission.findById(params?.submissionId)
-            submission.number = Integer.parseInt(params?.submissionNumber)
-            submission.type = params?.submissionType
-            submission.comments = params?.comments
-            if (!submission.save(flush: true)) {
-                flash.message = submission.errors.allErrors.collect { it }.join("<br/>")
+        JsonParser parser = new JsonParser()
+        JsonArray dataSubmission = parser.parse(request.parameterMap["submission"].toString())
+        List<MultipartFile> files = request.multiFileMap.collect { it.value }.flatten()
+        User user = getUser()
+
+        try {
+            Submission submission
+            if (params?.submissionId) {
+                submission = Submission.findById(params?.submissionId)
+                submission.comments = dataSubmission[0].comments.value
+                submission.type = dataSubmission[0].type.value
+            } else {
+                submission = getJson(Submission.class, dataSubmission[0])
+                submission.createDate = new Date()
+                submission.author = getUser()?.displayName
+                submission.documents = new ArrayList<StorageDocument>()
             }
-        } else {
-            submission = new Submission(
-                    projectKey: params.projectKey,
-                    number: Integer.parseInt(params?.submissionNumber),
-                    author: getUser()?.displayName,
-                    type: params?.submissionType,
-                    comments: params.comments,
-                    createDate: new Date(),
-                    documents: new ArrayList<StorageDocument>())
+            if(!files?.isEmpty()) {
```

Comment:
Even though you're looking at files here, the thing you're using is the `parts` part of the `request`. You might be able to iterate over `request.parts` instead.

Reply:
This seems not be working for iteration.

Reply:
I don't know what the exact right answer is here, only that accessing `request.multiFileMap` directly is brittle.

Reply:
OK... I'll try some ideas.